tests/integration/askPresentAndSquadJoin.test.ts

```diff
@@ -0,0 +1,219 @@
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+import { eq } from "drizzle-orm";
+import { afterEach, beforeAll, describe, expect, it, vi } from "vitest";
+
+import { presentIdeas, squadJoins, testItems, users } from "@/db/schema";
+import { constructPresentIdeaSavedMessage } from "@/services/slack/constructPresentIdeaSavedMessage";
+import { pollInterval, timeout, waitTimeout } from "@/testUtils/constants";
+import { deleteLastDm } from "@/testUtils/integration/deleteLastDm";
+import { sendScheduleEvent } from "@/testUtils/integration/sendScheduleEvent";
+import { sendSlackInteraction } from "@/testUtils/integration/sendSlackInteraction";
+import { waitForDm } from "@/testUtils/integration/waitForDm";
+import { testDb, waitForTestItem } from "@/testUtils/testDb";
+import { scheduleEvent } from "@/types/schedule";
+import {
+  additionalPresentIdeasInputActionId,
+  additionalPresentIdeasSaveButtonActionId,
+  squadJoinCheckboxActionId,
+} from "@/types/SlackInteractionRequest";
+
+dayjs.extend(utc);
+
+const constants = vi.hoisted(() => ({
+  teamId: "T1",
+  birthdayPerson: "U1",
+  userId: "U2",
+  presentIdea: "Test idea",
+}));
+
+describe("Present and Squad Join", () => {
+  beforeAll(async () => {
+    await testDb.delete(users);
+    await testDb.delete(testItems);
+    await testDb.delete(presentIdeas);
+  });
+
+  afterEach(async () => {
+    await deleteLastDm();
+
+    await testDb.delete(users);
+    await testDb.delete(testItems);
+    await testDb.delete(presentIdeas);
+  });
+
+  it(
+    "Should ask for present and squad join in DM",
+    async () => {
+      await testDb.insert(users).values([
+        {
+          id: import.meta.env.VITE_SLACK_USER_ID,
+          teamId: import.meta.env.VITE_SLACK_TEAM_ID,
+          birthday: new Date(),
+        },
+        {
+          id: constants.birthdayPerson,
+          teamId: import.meta.env.VITE_SLACK_TEAM_ID,
+          birthday: dayjs.utc().add(2, "month").toDate(),
```

Comment:
Does this have to be 2 months away? Its confusing to me because its supposed to be 2 months - 4 days.

Reply:
For here actually doesn't matter, because we passing the birthday person's id, so there is no additional calculation.